src/components/chips/js/chipsDirective.js

```diff
@@ -0,0 +1,86 @@
+(function () {
+  'use strict';
+  angular
+      .module('material.components.chips')
+      .directive('mdChips', MdChips);
+
+  /**
+   * @ngdoc directive
+   * @name mdChips
+   * @module material.components.chips
+   *
+   * @description
+   * `<md-chips>` 
+   *
+   */
+
+  function MdChips ($compile) {
+    return {
+      template: function(element, attrs) {
+        var userTemplate = element.clone();
```

Comment:
Here's a comment I wrote a while back explaining this technique:
```js
      // Specifying templateUrl as a function gives us the opportunity to
      // store the user-provided template for the directive instance, which
      // we can manually parse (rather than having angular compile them).
      // The attrs object is a conventient place to store this template because
      // the same attrs are passed to the appropriate link function invocation
      // for this directive instance. Prefix the property name with a $ so that
      // angular will not actually append the property to the rendered element.
      // The element is cloned because this element reference actually has its
      // contents replaced by the directive's rendered template.
```